otsdaq-mu2e/FEInterfaces/CFOFrontEndInterface_interface.cc

```diff
@@ -84,105 +87,155 @@ void CFOFrontEndInterface::registerFEMacros(void)
 
 	registerFEMacroFunction(
 		"CFO Write",  // feMacroName
 			static_cast<FEVInterface::frontEndMacroFunction_t>(
 					&CFOFrontEndInterface::WriteCFO),  // feMacroFunction
 					std::vector<std::string>{"address", "writeData"},
 					std::vector<std::string>{},  // namesOfOutput
-					1);                          // requiredUserPermissions
+					1,    // requiredUserPermissions
+					"*",  // allowedCallingFEs
+					"This FE Macro writes to the CFO registers."
+	);
 
 	registerFEMacroFunction(
 		"Loopback Test",  // feMacroName
 			static_cast<FEVInterface::frontEndMacroFunction_t>(
 					&CFOFrontEndInterface::LoopbackTest),  // feMacroFunction
 					std::vector<std::string>{"loopbacks", "link", "delay"},
 					std::vector<std::string>{"Response"},  // namesOfOutput
-					1); 
+					1,
+					"*", 
+					"Similar to <b>Test Loopback marker</b>, this FE Macro repeatedly measures the delay of markers from ROCs for a specified link. "
+					"The average delay is returned given the number of iterations (loopbacks), link, and delay (sleep between iterations). "
+					"This FE Macro is useful for Event Window synchronization.\n\n"
+					"Constraints:\n"
+					"\t-Loopback must be less than 10,000.\n"
+	); 
 
 	registerFEMacroFunction(
 		"Test Loopback marker",  // feMacroName
 			static_cast<FEVInterface::frontEndMacroFunction_t>(
 					&CFOFrontEndInterface::TestMarker),  // feMacroFunction
 					std::vector<std::string>{"DTC-chain link index (0-7)"},
 					std::vector<std::string>{"Response"},  // namesOfOutput
-					1); 
+					1,
+					"*",
+					"This FE Macro measures the delay of a marker from ROCs. "
+					"Optionally, the delay can be measured over mutliple iterations with the <b>Loopback Test</B> Macro."
+	); 
 
 	registerFEMacroFunction(
 		"CFO Read",
 			static_cast<FEVInterface::frontEndMacroFunction_t>(
 					&CFOFrontEndInterface::ReadCFO),                  // feMacroFunction
 					std::vector<std::string>{"address"},  // namesOfInputArgs
 					std::vector<std::string>{"readData"},
-					1);  // requiredUserPermissions
+					1,  // requiredUserPermissions
+					"*", 
+					"Read from the CFO Memory Map.\n\n"
+					"Parameters:\n"
+					"\taddress (uint16_t): Address in Memory Map.\n"
+	); 
 
 	registerFEMacroFunction(
 		"Select Jitter Attenuator Source",
 			static_cast<FEVInterface::frontEndMacroFunction_t>(
 					&CFOFrontEndInterface::SelectJitterAttenuatorSource),
 				        std::vector<std::string>{"Source (0 is Local oscillator, 1 is RTF Copper Clock)","DoNotSet"},
 						std::vector<std::string>{"Register Write Results"},
-					1);  // requiredUserPermissions
+					1, // requiredUserPermissions
+					"*",
+					"The Jitter Attenuator is used to remove jitter, or variation in the timing of signals. "
+					"Select the source of the jitter attenuator: a local oscilator on the CFO or RTF.\n"
+					"The RTF (RJ45 Timing Fanout) is a separate board to alleviate jitter accumulation. <b>Not all DTCs are connected to RTF</b>."
```

Comment:
connected to the RTF

Reply:
Updated JA tooltip in DTC and CFO interfaces